src/main/telemetry/esc_telemetry.c

```diff
@@ -0,0 +1,298 @@
+
+static uint16_t escVbat = 0;
+static uint16_t escCurrent = 0;
+static uint16_t escConsumption = 0;
+
+static void escTelemetryDataReceive(uint16_t c);
+static uint8_t update_crc8(uint8_t crc, uint8_t crc_seed);
+static uint8_t get_crc8(uint8_t *Buf, uint8_t BufLen);
+static void selectNextMotor(void);
+
+bool isEscTelemetryActive(void)
+{
+    return escTelemetryEnabled;
+}
+
+uint16_t getEscTelemetryVbat(void)
+{
+    return escVbat / 10;
+}
+
+uint16_t getEscTelemetryCurrent(void)
+{
+    return escCurrent;
+}
+
+uint16_t getEscTelemetryConsumption(void)
+{
+    return escConsumption;
+}
+
+bool escTelemetryInit(void)
+{
+    serialPortConfig_t *portConfig = findSerialPortConfig(FUNCTION_TELEMETRY_ESC);
+    if (!portConfig) {
+        return false;
+    }
+
+    portOptions_t options = (SERIAL_NOT_INVERTED);
+
+    // Initialize serial port
+    escTelemetryPort = openSerialPort(portConfig->identifier, FUNCTION_TELEMETRY_ESC, escTelemetryDataReceive, ESC_TLM_BAUDRATE, MODE_RX, options);
+
+    if (escTelemetryPort) {
+        escTelemetryEnabled = true;
+        masterConfig.batteryConfig.currentMeterType = CURRENT_SENSOR_ESC;
+        masterConfig.batteryConfig.batteryMeterType = BATTERY_SENSOR_ESC;
+    }
+
+    return escTelemetryPort != NULL;
+}
+
+void freeEscTelemetryPort(void)
+{
+    closeSerialPort(escTelemetryPort);
+    escTelemetryPort = NULL;
+    escTelemetryEnabled = false;
+}
+
+// Receive ISR callback
+static void escTelemetryDataReceive(uint16_t c)
+{
+    // KISS ESC sends some data during startup, ignore this for now (maybe future use)
+    // startup data could be firmware version and serialnumber
+
+    if (escTelemetryTriggerState == ESC_TLM_TRIGGER_WAIT) return;
+
+    tlm[tlmFramePosition] = (uint8_t)c;
+
+    if (tlmFramePosition == ESC_TLM_BUFFSIZE - 1) {
+        tlmFrameDone = true;
+        tlmFramePosition = 0;
+    } else {
+        tlmFramePosition++;
+    }
+}
+
+uint8_t escTelemetryFrameStatus(void)
+{
+    uint8_t frameStatus = ESC_TLM_FRAME_PENDING;
+    uint16_t chksum, tlmsum;
+
+    if (!tlmFrameDone) {
+        return frameStatus;
+    }
+
+    tlmFrameDone = false;
+
+    // Get CRC8 checksum
+    chksum = get_crc8(tlm, ESC_TLM_BUFFSIZE - 1);
+    tlmsum = tlm[ESC_TLM_BUFFSIZE - 1];     // last byte contains CRC value
+
+    if (chksum == tlmsum) {
+        escTelemetryData[escTelemetryMotor].skipped = false;
+        escTelemetryData[escTelemetryMotor].temperature = tlm[0];
+        escTelemetryData[escTelemetryMotor].voltage = tlm[1] << 8 | tlm[2];
+        escTelemetryData[escTelemetryMotor].current = tlm[3] << 8 | tlm[4];
+        escTelemetryData[escTelemetryMotor].consumption = tlm[5] << 8 | tlm[6];
+        escTelemetryData[escTelemetryMotor].rpm = tlm[7] << 8 | tlm[8];
+
+        frameStatus = ESC_TLM_FRAME_COMPLETE;
+    }
+
+    return frameStatus;
+}
+
+void escTelemetryProcess(uint32_t currentTime)
+{
+    UNUSED(currentTime);
+
```

Comment:
Set new variable `currentTimeMs` to `currentTime / 1000` and use that rather than repeatedly calling `millis()`